doc/source/notebooks/advanced/het_gpr_script.ipynb

```diff
@@ -0,0 +1,512 @@
+      "text/plain": [
+       "<Figure size 1080x360 with 1 Axes>"
+      ]
+     },
+     "metadata": {
+      "needs_background": "light"
+     },
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "mean, var = base_model.predict_y(X)\n",
+    "plot_distribution(X, Y, mean.numpy(), np.sqrt(var.numpy()))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "a55d7173",
+   "metadata": {},
+   "source": [
+    "### Constructing a custom likelihood for a heteroskedastic GPR Model"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "5d803b66",
+   "metadata": {},
+   "source": [
+    "This likelihood class allows the noise to vary linearly with the input location."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "id": "f5e96f63",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "class LinearLikelihood(MultiLatentTFPConditional):\n",
+    "\n",
+    "    def __init__(self, ndims: int = 1,  **kwargs):\n",
+    "        self.noise_gradient = Parameter(np.ones(ndims), prior=GRADIENT_PRIOR)\n",
+    "        self.constant_noise = Parameter(1.0, transform=positive(lower=MIN_NOISE_SCALE))\n",
```

Comment:
I'm not sure how much it really makes sense to constrain the constant to be positive? With the gradient, the scaled value could be very different from the constant? Maybe you should be doing something down in `scale_transform` to ensure positiveness instead?